src/github_repo_request_local.py

```diff
@@ -83,15 +162,55 @@ def list_test_files(directory, excluded_extensions):
     else:
         logger.error(f"CSV file not found at {csv_file_path}.")
 
+# Filter out rows where the URL doesn't have a repository name (83 rows)
+if "repourl" in df.columns:
+    # Identify rows with incomplete URLs
+    incomplete_urls = df[
+        df["repourl"].apply(lambda x: len(x.rstrip("/").split("/")) < 5)
+    ]
+
+    # Log the incomplete URLs
+    if not incomplete_urls.empty:
+        logger.info("Incomplete GitHub URLs found and will be excluded:")
+        for url in incomplete_urls["repourl"]:
+            logger.info(f"Excluding the repourl : {url}")
+
+    df = df[df["repourl"].apply(lambda x: len(x.rstrip("/").split("/")) >= 5)]
+
+
+if "last_commit_hash" not in df.columns:
+    df["last_commit_hash"] = None  # Initialize the column with None
+
+# replace http with https
+df["repourl"] = df["repourl"].str.replace(r"^http\b", "https", regex=True)
+
+# Some of the URLs end with "/". I need to remove them.
+df["repourl"] = df["repourl"].str.rstrip("/")
+
+# Clean and filter URLs
+df["repourl"] = df["repourl"].apply(get_base_repo_url)
+df = df[df["repourl"] != ""]  # Remove any rows with invalid URLs
+
+# Number of repositories to process before saving to CSV
+BATCH_SIZE = 10
+
+# Track the number of processed repositories in the current batch
+processed_count = 0
 
 for index, row in df.iterrows():
-    if row["testfilecountlocal"] != -1:
-        continue  # Skip processed repositories
+    # Check if we need to skip this repository because it's fully processed
+    if row["testfilecountlocal"] != -1 and pd.notna(row["last_commit_hash"]):
+        continue
 
-    repo_url = row["repourl"]
+    original_repo_url = row["repourl"]
+    repo_url = get_base_repo_url(original_repo_url)
+    if not repo_url:
+        logger.error(f"Invalid repository URL: {original_repo_url}")
```

Comment:
This seems more like a warning than an error. Why do you think an error is the most appropriate log level? Where's the error occurred - in our code's processing? if so, then I'd agree this an error log entry might be the most appropriate choice. If the error's related to the input data then we've already accepted that some of the entries aren't valid as-is and clean them up or remove them; and we log these using `logger.info(...)`